### PR TITLE
Ports colored chat names

### PR DESCRIPTION
**Pull request by ike709**
## About the PR

Ports https://github.com/space-syndicate/space-station-14/pull/844

I think this makes it much easier to follow what people are saying, and when we have the technology for runechat that looks good we can color it there like in SS13.

I'm pretty indifferent about what the actual color options are so bikeshed as much as you please.

## Media

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Morty, ike709
- add: Names are now colored in chat.

### CODE REVIEW

**Comment by Kadeo64** [comment]
do we get to choose our chat color or nah

**Comment by mirrorcult** [comment]
how does it work with identity rn

**Comment by mirrorcult** [comment]
> do we get to choose our chat color or nah

dont think its necessary imo

**Comment by mirrorcult** [comment]
at least not for initial pr

File: Content.Shared/Humanoid/SharedHumanoidAppearanceSystem.cs [hunk 1]
```diff
@@ -328,6 +329,11 @@ public virtual void LoadProfile(EntityUid uid, HumanoidCharacterProfile profile,
 
         humanoid.Age = profile.Age;
 
+        const string paletteId = "Material";
```
Comment (metalgearsloth) [inline Content.Shared/Humanoid/SharedHumanoidAppearanceSystem.cs:332]: ValidatePrototypeId attribute exists if you want to hardcode a prototype.

**Comment by ike709** [comment]
> how does it work with identity rn

Here's how it works now with a voice changer:

image

**Comment by deathride58** [comment]
Does this color the speech bubble name as well?

**Comment by Emisse** [comment]
im going senile i cant read chat its all white i have to merge this now

**Comment by Errant-4** [comment]
Nice!
But is it intentional that department channel color overwrites the name color? Names in emotes are also not colored, is that because this is supposed to represent voice only?

**Comment by SlamBamActionman** [comment]
This was merged before proper testing, and the screenshot in the PR does not accurately reflect how it looks ingame. There is a thread on Discord with multiple examples of how the current color selection is detrimental to the readability of chat. https://discord.com/channels/310555209753690112/1200030805095895050

Revert this PR or get a hotfix out to fix the colors/make it optional.

**Comment by SlamBamActionman** [comment]
image-7

One of the examples. Ite Dino's first message is the worst blended, but even still the others require more effort to discern at a glance.

**Comment by GreyMario** [comment]
very accessible thanks i'm so glad i'm not colorblind

**Comment by SlamBamActionman** [comment]
For those that need visuals for the comment above, here are examples of various color blindness using the edited bg from #24555:

Original:
299670854-3ed313c5-07ce-40c0-b75f-7be4dda776c2

Deutan:
pilestone jpg (3)

Protan:
pilestone jpg (4)

Tritan:
pilestone jpg (5)


I also recommend to try reading the text while squinting; it can help determine if certain color contrasts blend in too much in the background.

**Comment by juliangiebel** [comment]
As Ike mentioned here:
> I'm pretty indifferent about what the actual color options are so bikeshed as much as you please.

The colors aren't specifically chosen so feel free to open a PR that sets the colors to an accessible color pallet. Posting additional comments on this PR won't help and eventually just get lost.

**Pull request merged**